libs/clients/university-application/university-of-iceland/src/lib/apiConfiguration.ts

```diff
@@ -0,0 +1,86 @@
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+import {
+  ConfigType,
+  IdsClientConfig,
+  XRoadConfig,
+} from '@island.is/nest/config'
+import { Configuration, CoursesApi, ProgramsApi } from '../../gen/fetch'
+import { UniversityOfIcelandApplicationClientConfig } from './universityOfIcelandClient.config'
+
+const configFactory = (
+  xRoadConfig: ConfigType<typeof XRoadConfig>,
+  config: ConfigType<typeof UniversityOfIcelandApplicationClientConfig>,
+  idsClientConfig: ConfigType<typeof IdsClientConfig>,
+  basePath: string,
+) => ({
+  fetchApi: createEnhancedFetch({
+    name: 'clients-university-application-university-of-iceland',
+    autoAuth: idsClientConfig.isConfigured
+      ? {
+          mode: 'auto',
+          issuer: idsClientConfig.issuer,
+          clientId: idsClientConfig.clientId,
+          clientSecret: idsClientConfig.clientSecret,
+          scope: config.scope,
+        }
+      : undefined,
+    timeout: config.fetchTimeout,
+  }),
+  headers: {
+    'X-Road-Client': xRoadConfig.xRoadClient,
+    'Content-Type': 'application/json',
+    Accept: 'application/json',
+  },
+  basePath,
+})
+
+export const exportedApis = [
```

Comment:
To limit duplication you can have a look at [this](https://github.com/island-is/island.is/blob/main/libs/clients/fishing-license/src/lib/apis.ts) where the configuration is reused for the apis exported by the client module.